.github/workflows/ci.yaml

```diff
@@ -0,0 +1,78 @@
+name: CI
+
+on:
+  push:
+    branches: ['main']
+  pull_request:
+    types: [ opened, synchronize, reopened ]
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: ${{ github.ref != 'refs/heads/main' }}
+
+jobs:
```

Comment:
Do we have a reference or a template of this file? I'm not familiar with most of these fields.

Reply:
@CCisGG updated PR description with details.